src/libs/TransactionUtils/index.ts

```diff
@@ -152,6 +154,18 @@ function buildOptimisticTransaction(
         commentJSON.originalTransactionID = originalTransactionID;
     }
 
+    const isDistanceTransaction = !!pendingFields?.waypoints;
+    if (isDistanceTransaction) {
+        // Get the policy of this transaction from the report.policyID
+        const allReports = ReportConnection.getAllReports();
+        const report = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] ?? null;
+        const policyID = report?.policyID ?? '';
+        const policy = PolicyUtils.getPolicy(policyID);
```

Comment:
The IOU report may not exist (if you are creating a new IOU report i.e. no other outstanding expenses). This will result in an `undefined` policy and in turn in an empty `mileageRates` object causing this bug https://github.com/Expensify/App/pull/50001#issuecomment-2414991341. Let's pass the `policy` instead.

Reply:
Everywhere this is called we have an optimistic report, except for with distance splits, but then it uses the P2P rate so an undefined policy is fine.

I'm not sure how there could be a problem here or what flow would hit it.

Reply:
We have an optimistic report but when this code is being executed the optimistic report is still in memory (just a variable) and not transmitted to onyx yet (the Onyx.update call is not made yet). Thus if you try to get that report from onyx you won't get it

Reply:
Ah right, that's the part I was missing! Thanks for catching this!